pa-rserve/src/main/java/org/ow2/parserve/PARServeEngine.java

```diff
@@ -0,0 +1,446 @@
+
+            if (!serverEval) {
+                // PRC-32 A ScriptException() must be thrown if the script calls stop() function
+                ScriptException toThrow = null;
+                if (listener.lastErrorMessage != null) {
+                    toThrow = new ScriptException(listener.lastErrorMessage);
+                }
+                if (toThrow != null) {
+                    throw toThrow;
+                }
+            }
+        }
+    }
+
+    /**
+     * Initialize the Tailer thread used to read R output
+     */
+    private void initializeTailer(Bindings bindings, ScriptContext ctx) throws ScriptException {
+        if (!serverEval) {
+            Tailer tailer = null;
+            outputFile = createOuputFile(bindings);
+
+            listener = new PARScriptTailerListener(ctx.getWriter(), this);
+            tailer = new Tailer(outputFile, listener, TAILER_PERIOD);
+
+            engine.initializeOutput(outputFile, ctx);
+
+            tailerThread = new Thread(tailer, "PARServeEngine Tailer");
+            tailerThread.setDaemon(true);
+            tailerThread.start();
+        }
+    }
+
+    /**
+     * Terminate the Tailer thread used to read R output
+     */
+    private void terminateTailer() {
+        if (!serverEval) {
+            try {
+                tailerThread.join(TAILER_TIMEOUT);
+            } catch (InterruptedException e) {
+
+            }
+            if (tailerThread.isAlive()) {
+                tailerThread.interrupt();
+            }
+            if (outputFile.exists()) {
+                outputFile.delete();
+            }
+        }
+    }
+
+    @Override
+    public Object eval(Reader reader, ScriptContext context) throws ScriptException {
+        String s;
+        try {
+            s = CharStreams.toString(reader);
+        } catch (IOException ex) {
+            throw new ScriptException(ex);
+        }
+        return eval(s, context);
+    }
+
+
+    /**
+     * This class reads the content of the .Rout file produced by the Rsession as the tail command.
+     */
+    public class PARScriptTailerListener extends TailerListenerAdapter {
+
+        Writer writer;
+
+        String lastErrorMessage;
+
+        Tailer tailer;
+
+        PARServeEngine engine;
+
+        boolean readError = false;
+
+        StringBuilder error = new StringBuilder();
+
+        public PARScriptTailerListener(Writer wr, PARServeEngine engine) {
+            this.writer = wr;
+            this.engine = engine;
+        }
+
+        @Override
+        public void handle(Exception ex) {
+            if (ex instanceof InterruptedIOException) {
+                try {
+                    writer.close();
+                } catch (IOException e) {
+
+                }
+            }
+            super.handle(ex);
+        }
+
+        @Override
+        public void init(Tailer tailer) {
+            super.init(tailer);
+            this.tailer = tailer;
+        }
+
+        public void handle(String line) {
```

Comment:
I will see what I can do. Rserve log handling (tail file) is different from JRI (callbacks)